ruby/gilded_rose.rb

```diff
@@ -65,4 +24,128 @@ def initialize(name, sell_in, quality)
   def to_s()
     "#{@name}, #{@sell_in}, #{@quality}"
   end
+end
+
+class ItemUpdaterFactory
+  def self.create_updater_for(item)
+    { "Sulfuras, Hand of Ragnaros" => LegendaryItemUpdater.new(item),
+      "Aged Brie" => AgedBrieItemUpdater.new(item),
+      "Backstage passes to a TAFKAL80ETC concert" =>
+        BackstagePassesItemUpdater.new(item),
+      "Conjured" => ConjuredItemUpdater.new(item)
+    }.fetch(lookup_name(item), ItemUpdater.new(item))
+  end
+
+  def self.lookup_name(item)
+    if item.name.include?("Conjured")
+      "Conjured"
+    else
+      item.name
+    end
+  end
+end
+
+module Updater
+  attr_accessor :item
+
+  def initialize (item)
+    @item = item
+  end
+
+  def sell_in
+    item.sell_in
+  end
+
+  def quality
+    item.quality
+  end
+
+  def increase_quality

```

Comment:
A Rubyism is adding a bang to methods that have side-effects like mutation. So, `increase_quality!` might be a better name.

A warning, though: Ruby devs go overboard with them, so use sparingly. 😆

Reply:
The side-effect-iness of this solution in general feels kinda odd to me. I mostly kept the original code and functionality and just put it in different places, so I didn't try to make it non-side-effecting. As it is, I think just about every method needs a `!`!